### PR TITLE
[java] Fills out the javadoc so there are no more documentation warnings

### PR DESCRIPTION
**Pull request by Craigacp**
### Description
Adds javadoc for all protected and public members, methods and classes.

### Motivation and Context
The javadoc warnings were annoying me when running the builds. Also, those types should have been documented.

### CODE REVIEW

File: java/src/main/java/ai/onnxruntime/TensorInfo.java [hunk 1]
```diff
@@ -16,24 +16,41 @@ public class TensorInfo implements ValueInfo {
...
     ONNX_TENSOR_ELEMENT_DATA_TYPE_BOOL(13),
+    /** A 64-bit complex number, stored as 2 32-bit values. Not accessible from Java. */
     ONNX_TENSOR_ELEMENT_DATA_TYPE_COMPLEX64(
         14), // complex with float32 real and imaginary components
+    /** A 128-bit complex number, stored as 2 64-bit values. Not accessible from Java. */
     ONNX_TENSOR_ELEMENT_DATA_TYPE_COMPLEX128(
         15), // complex with float64 real and imaginary components
+    /** A non-IEEE 16-bit floating point value. */
     ONNX_TENSOR_ELEMENT_DATA_TYPE_BFLOAT16(
         16); // Non-IEEE floating-point format based on IEEE754 single-precision
```
Comment (skottmckay) [inline java/src/main/java/ai/onnxruntime/TensorInfo.java:55]: Could maybe link to https://en.wikipedia.org/wiki/Bfloat16_floating-point_format to provide details.

File: java/src/main/java/ai/onnxruntime/providers/CoreMLFlags.java [hunk 2]
```diff
@@ -1,15 +1,19 @@
 /*
- * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2021, 2023, Oracle and/or its affiliates. All rights reserved.
  * Licensed under the MIT License.
  */
 package ai.onnxruntime.providers;
 
 /** Flags for the CoreML provider. */
 public enum CoreMLFlags implements OrtFlags {
+  /** Use only the CPU, disables the GPU and Apple Neural Engine. */
```
Comment (skottmckay) [inline java/src/main/java/ai/onnxruntime/providers/CoreMLFlags.java:9]:
```suggestion
  /** Use only the CPU, disables the GPU and Apple Neural Engine. 
      Only recommended for developer usage as it significantly impacts performance.
   */
```

File: java/src/main/java/ai/onnxruntime/providers/CoreMLFlags.java [hunk 3]
```diff
@@ -1,15 +1,19 @@
...
   CPU_ONLY(1), // COREML_FLAG_USE_CPU_ONLY(0x001)
+  /** Enables CoreML on subgraphs. */
   ENABLE_ON_SUBGRAPH(2), // COREML_FLAG_ENABLE_ON_SUBGRAPH(0x002)
+  /** Only enables the Apple Neural Engine. */
```
Comment (skottmckay) [inline java/src/main/java/ai/onnxruntime/providers/CoreMLFlags.java:13]:
```suggestion
  /** Only enable usage of CoreML if the device has an Apple Neural Engine. */
```

File: java/src/main/java/ai/onnxruntime/providers/NNAPIFlags.java [hunk 4]
```diff
@@ -1,16 +1,21 @@
...
+ * Copyright (c) 2021, 2023, Oracle and/or its affiliates. All rights reserved.
  * Licensed under the MIT License.
  */
 package ai.onnxruntime.providers;
 
 /** Flags for the NNAPI provider. */
 public enum NNAPIFlags implements OrtFlags {
+  /** Enables fp16 support. */
   USE_FP16(1), // NNAPI_FLAG_USE_FP16(0x001)
+  /** Uses channels first format. */
```
Comment (skottmckay) [inline java/src/main/java/ai/onnxruntime/providers/NNAPIFlags.java:11]:
```suggestion
  /** Uses channels first format. 
      Only recommended for developer usage to validate code changes to the execution provider implementation.
  */
```

File: java/src/main/java/ai/onnxruntime/providers/NNAPIFlags.java [hunk 5]
```diff
@@ -1,16 +1,21 @@
...
   USE_NCHW(2), // NNAPI_FLAG_USE_NCHW(0x002)
+  /** Disables CPU ops. */
```
Comment (skottmckay) [inline java/src/main/java/ai/onnxruntime/providers/NNAPIFlags.java:13]:
```suggestion
  /** Disables NNAPI from using CPU. 
      If an operator could be assigned to NNAPI, but NNAPI only has a CPU implementation of that operator 
      on the current device, model load will fail. */
```

File: java/src/main/java/ai/onnxruntime/providers/NNAPIFlags.java [hunk 6]
```diff
@@ -1,16 +1,21 @@
...
   CPU_DISABLED(4), // NNAPI_FLAG_CPU_DISABLED(0x004)
+  /** Only runs on CPU. */
```
Comment (skottmckay) [inline java/src/main/java/ai/onnxruntime/providers/NNAPIFlags.java:15]:
```suggestion
  /** NNAPI will only use CPU. 
      Only recommended for developer usage as it significantly impacts performance.
  */
```

File: java/src/main/java/ai/onnxruntime/OrtLoggingLevel.java [hunk 7]
```diff
@@ -8,10 +8,15 @@
 
 /** The logging level for messages from the environment and session. */
```
[the remark this answers is not in the document]
Comment (Craigacp) [reply java/src/main/java/ai/onnxruntime/OrtLoggingLevel.java:9]: `OrtLoggingLevel` is used for the global logging level and the per session & run ones in the Java API. The methods that set the logging levels (e.g. https://github.com/microsoft/onnxruntime/blob/main/java/src/main/java/ai/onnxruntime/OrtEnvironment.java#L82, https://github.com/microsoft/onnxruntime/blob/main/java/src/main/java/ai/onnxruntime/OrtSession.java#L696, https://github.com/microsoft/onnxruntime/blob/main/java/src/main/java/ai/onnxruntime/OrtSession.java#L1266) note that they are for the object they are on, but it's probably not clear how they override each other.

File: java/src/main/java/ai/onnxruntime/TensorInfo.java (hunk 1 above)
Comment (Craigacp) [reply java/src/main/java/ai/onnxruntime/TensorInfo.java]: Fixed.

File: java/src/main/java/ai/onnxruntime/OrtSession.java [hunk 8]
```diff
@@ -499,7 +509,9 @@ public int getID() {
      * The execution mode to use. Needs to be kept in sync with the ExecutionMode enum in the C API.
      */
     public enum ExecutionMode {
+      /** Executes all nodes sequentially. */
```
[the remark this answers is not in the document]
Comment (Craigacp) [reply java/src/main/java/ai/onnxruntime/OrtSession.java]: Fixed.

File: java/src/main/java/ai/onnxruntime/OrtSession.java [hunk 9]
```diff
@@ -474,9 +474,19 @@ public static class SessionOptions implements AutoCloseable {
      * in the C API.
      */
     public enum OptLevel {
```
[the remark this answers is not in the document]
Comment (Craigacp) [reply java/src/main/java/ai/onnxruntime/OrtSession.java:480]: Fixed.

**Comment by Craigacp** [comment]
The Windows CPU CI run failed with error 500 when trying to download node and the TVM run failed with a cpptest error, neither of which seem related to this change. Maybe the Windows CPU CI one is transient?

**Review by skottmckay — APPROVED** [review]

**Pull request merged**